dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/NotificationRuleActionImplementer.java

```diff
@@ -86,7 +89,8 @@ protected ExternalNotificationLogEntry createLogEntry(String key, String templat
         return entry;
     }
 
-    protected ProgramNotificationTemplate getNotificationTemplate( RuleAction action )
+    @Transactional
```

Comment:
Consider removing this annotation. The method doesn't accces the DB directly, but only through another method (`getNotificationTemplate`) which is already annotated with `@Transactional`

Reply:
fixed